src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -926,14 +1056,35 @@ def option(self, key: str, value: Any, _emit_ast: bool = True) -> "DataFrameRead
         """Sets the specified option in the DataFrameReader.
 
         Use this method to configure any
-        `format-specific options <https://docs.snowflake.com/en/sql-reference/sql/create-file-format.html#format-type-options-formattypeoptions>`_
-        and
-        `copy options <https://docs.snowflake.com/en/sql-reference/sql/copy-into-table.html#copy-options-copyoptions>`_.
+        `format-specific options <https://docs.snowflake.com/en/sql-reference/sql/create-file-format.html#format-type-options-formattypeoptions>`_,
+        `copy options <https://docs.snowflake.com/en/sql-reference/sql/copy-into-table.html#copy-options-copyoptions>`_,
+        and `time travel options <https://docs.snowflake.com/en/sql-reference/constructs/at-before>`_.
         (Note that although specifying copy options can make error handling more robust during the
         reading process, it may have an effect on performance.)
 
+        Time travel options can be used with ``table()`` method:
+            - ``time_travel_mode``: Either "at" or "before"
+            - ``statement``: Query ID for statement-based time travel
+            - ``offset``: Seconds to go back in time (negative integer)
+            - ``timestamp``: Specific timestamp for time travel
+            - ``timestamp_type``: Type of timestamp interpretation ('NTZ', 'LTZ', or 'TZ').
+
+              Smart timestamp_type handling:
+
+              - datetime with timezone + timestamp_type NOT provided → auto-sets to 'TZ'
+              - datetime with timezone + timestamp_type explicitly provided → uses provided timestamp_type
+              - datetime without timezone (naive) + timestamp_type NOT provided → no casting, raw string (timestamp_type=None)
+              - datetime without timezone (naive) + timestamp_type explicitly provided → uses provided timestamp_type
+              - string timestamps + timestamp_type NOT provided → no casting, raw string (timestamp_type=None, Snowflake native handling)
+              - string timestamps + timestamp_type explicitly provided → uses TO_TIMESTAMP_XXX casting
```

Comment:
this looks like internal implementation details, I think we can exclude it first and add back based upon user feedback